src/lib/provable/string.ts

```diff
@@ -42,24 +42,27 @@ class Character extends Struct({ value: Field }) {
   }
 }
 
+const stringEncoder = new TextEncoder();
+
+const stringDecoder = new TextDecoder('utf-8');
+
 // construct a provable with a `string` js type
 const RawCircuitString = {
   ...provable({ values: Provable.Array(Character, DEFAULT_STRING_LENGTH) }),
 
   toValue({ values }) {
-    return values
-      .map((x) => x.toString())
-      .join('')
-      .replace(/[^ -~]+/g, '');
+    return stringDecoder.decode(
+      new Uint8Array(values.map((x) => parseInt(x.toField().toString(), 10)))
```

Comment:
nit: I'd stick with numbers here instead of going to decimal strings and back
```suggestion
      new Uint8Array(values.map((x) => Number(x.toField().toBigInt()))
```